visualization/dashboard/src/MLIDashboard/Controls/FeatureImportanceBar/FeatureImportanceBar.tsx

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import _ from "lodash";
+import { localization } from "../../../Localization/localization";
+import { mergeStyleSets, getTheme } from "@uifabric/styling";
+import { ModelExplanationUtils } from "../../ModelExplanationUtils";
+import { IPlotlyProperty, AccessibleChart } from "mlchartlib";
+import { SpinButton } from "office-ui-fabric-react/lib/SpinButton";
+import { Slider } from "office-ui-fabric-react/lib/Slider";
+import { LoadingSpinner } from "../../SharedComponents";
+import { isThisExpression } from "@babel/types";
+import { Dropdown, IDropdownOption } from "office-ui-fabric-react/lib/Dropdown";
+import { InteractiveLegend, ILegendItem, SortingState } from "../InteractiveLegend";
+import { FabricStyles } from "../../FabricStyles";
+import { JointDataset } from "../../JointDataset";
+import { IGlobalSeries } from "../GlobalExplanationTab/IGlobalSeries";
+import { featureImportanceBarStyles } from "./FeatureImportanceBar.styles";
+import { Text } from "office-ui-fabric-react";
+
+export interface IFeatureBarProps {
+    jointDataset: JointDataset;
+    yAxisLabels: string[];
+    sortArray: number[];
+    selectedFeatureIndex?: number;
+    selectedSeriesIndex?: number;
+    topK: number;
+    startingK: number;
+    unsortedX: string[];
+    unsortedSeries: IGlobalSeries[];
+    onFeatureSelection?: (seriesIndex: number, featureIndex: number) => void;
+}
+
+export interface IFeatureBarState {
+    plotlyProps: IPlotlyProperty;
+}
+
+export class FeatureImportanceBar extends React.PureComponent<IFeatureBarProps, IFeatureBarState> {
+
+    constructor(props: IFeatureBarProps) {
+        super(props);
+        this.state = {
+            plotlyProps: undefined
+        };
+        this.selectPointFromChart = this.selectPointFromChart.bind(this )
+    }
+
+    public componentDidUpdate(prevProps: IFeatureBarProps) {
+        if (this.props.unsortedSeries !== prevProps.unsortedSeries || this.props.sortArray !== prevProps.sortArray) {
+            this.setState({plotlyProps: undefined});
+        }
+    }
+
+    public render(): React.ReactNode {
+        const classNames = featureImportanceBarStyles();
+        const relayoutArg = {'xaxis.range': [this.props.startingK - 0.5, this.props.startingK + this.props.topK - 0.5]};
+        const plotlyProps = this.state.plotlyProps;
+        _.set(plotlyProps, 'layout.xaxis.range', [this.props.startingK - 0.5, this.props.startingK + this.props.topK - 0.5]);
+
+        if (!this.props.unsortedSeries ||this.props.unsortedSeries.length === 0 || !this.props.sortArray || this.props.sortArray.length === 0) {
+            return (<div className={classNames.noData}>
+                <Text variant={"xxLarge"}>No data</Text>
+            </div>)
+        }
+        if (this.state.plotlyProps === undefined) {
+            this.loadProps();
+            return <LoadingSpinner/>;
```

Comment:
curious how does this work?  Does render get called again at some point and the logic gets rerun to remove the spinner?